### PR TITLE
Make the DF happy again

### PR DESCRIPTION
**Pull request by huwshimi**
Update the deployment flow to use the new controller API calls and fix a bunch of bugs to allow opening the DF again.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a service
- click deploy
- the deployment flow should open

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js [hunk 1]
```diff
@@ -89,26 +89,28 @@ YUI.add('deployment-credential-add', function() {
         return;
       }
       // Add template
-      var user = this.props.users.jem.user;
+      var user = this.props.user;
       var templateName = this.refs.templateName.getValue();
       var template = this._generateTemplate();
-      this.props.addTemplate(
-        user, templateName, template, this._addTemplateCallback);
+      // TODO: update this call to pass the correct parameters for the new API.
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js]: Card added.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5765/
Test PASSed.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/credential.js [hunk 2]
```diff
@@ -118,6 +105,7 @@ YUI.add('deployment-credential', function() {
         console.error('Unable to get credentials', error);
         return;
       }
+      console.log(credentials);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/credential.js]: Oops this was left in

**Comment by hatched** [comment]
Thanks @huwshimi One trivial, and I'll follow-up with a qa shortly

**Comment by hatched** [comment]
Thanks @huwshimi 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**